android/app/src/main/java/org/smartregister/fhircore/fragment/PatientDetailFragment.kt

```diff
@@ -70,76 +71,37 @@ class PatientDetailFragment : Fragment() {
         )
         .get(PatientListViewModel::class.java)
 
-    // patient id must be supplied
-    patientId = arguments?.getString(ARG_ITEM_ID)!!
-    viewModel.getPatientItem(patientId).observe(viewLifecycleOwner, { setupPatientData(it) })
-
     // load immunization data
-    viewModel
-      .searchImmunizations(patientId)
-      .observe(
-        viewLifecycleOwner,
-        {
-          if (it.isNotEmpty()) {
-            updateVaccineStatus(it)
-          }
-        }
-      )
+    loadProfile()
 
     return rootView
   }
 
-  private fun updateVaccineStatus(immunizations: List<Immunization>) {
-
-    var isFullyVaccinated = false
-    immunizations.forEach { immunization ->
-      doseNumber = (immunization.protocolApplied[0].doseNumber as PositiveIntType).value
-      initialDose = immunization.vaccineCode.coding.first().code
-      if (isFullyVaccinated) {
-        return@forEach
-      }
-      val nextDoseNumber = doseNumber + 1
-      val vaccineDate = immunization.occurrenceDateTimeType.toHumanDisplay()
-      val nextVaccineDate = Utils.addDays(vaccineDate, 28)
-      val tvVaccineRecorded = rootView.findViewById<TextView>(R.id.vaccination_status)
-      tvVaccineRecorded.text =
-        resources.getString(
-          R.string.immunization_brief_text,
-          immunization.vaccineCode.text,
-          doseNumber
-        )
+  private fun loadProfile() {
+    // bind patient summary
+    viewModel.getPatientItem(patientId).observe(viewLifecycleOwner, { setupPatientData(it) })
 
-      val tvVaccineSecondDose = rootView.findViewById<TextView>(R.id.vaccination_second_dose)
-      val btnRecordVaccine = activity?.findViewById<Button>(R.id.btn_record_vaccine)
-      tvVaccineSecondDose.visibility = View.VISIBLE
-      if (doseNumber == finalDoseNumber) {
-        isFullyVaccinated = true
-        tvVaccineRecorded.text = resources.getString(R.string.fully_vaccinated)
-        tvVaccineSecondDose.text = resources.getString(R.string.view_vaccine_certificate)
-        if (btnRecordVaccine != null) {
-          btnRecordVaccine.visibility = View.GONE
-        }
-      } else {
-        tvVaccineSecondDose.text =
-          resources.getString(R.string.immunization_next_dose_text, nextDoseNumber, nextVaccineDate)
-      }
-    }
+    // bind patient details
+    viewModel
+      .fetchPatientDetailsCards(requireContext(), patientId)
+      .observe(viewLifecycleOwner, { adapter.submitList(it) })
   }
 
   override fun onResume() {
     super.onResume()
-    // load immunization data
-    viewModel.searchImmunizations(patientId)
+    loadProfile()
   }
 
-  private fun setupPatientData(patient: PatientListViewModel.PatientItem?) {
+  private fun setupPatientData(patient: PatientItem?) {
     val gender = if (patient?.gender == "male") 'M' else 'F'
```

Comment:
additional refactor, can't this call `getPatientDemographics`? https://github.com/opensrp/fhircore/pull/388/files#diff-e83ec8d3d1e8ef0e815ec695937e105889ff938240a433b0e359e29b6c7ccdc7R98

Reply:
The methods achieve slightly different results. `getPatientDemographics` is in the `PatientItemViewHolder`

Reply:
yea I get that, not to replace this w/that but to call `getPatientDemographics` in this function, eg so we don't dup the abbreviating gender logic

Reply:
Agreed. Thanks for this